extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java

```diff
@@ -171,6 +176,36 @@ public AgroalDataSource doCreateDataSource(String dataSourceName) {
         if (dataSourceSupport.disableSslSupport) {
             agroalConnectionConfigurer.disableSslSupport(resolvedDbKind, dataSourceConfiguration);
         }
+        //we use a custom cache for two reasons:
+        //fast thread local cache should be faster
+        //and it prevents a thread local leak
+        dataSourceConfiguration.connectionPoolConfiguration().connectionCache(new ConnectionCache() {
+
+            FastThreadLocal<Acquirable> connectionCache = new FastThreadLocal<>();
+
+            @Override
+            public Acquirable get() {
+                Thread thread = Thread.currentThread();
+                if (thread instanceof FastThreadLocalThread || thread instanceof JBossThread) {
```

Comment:
I'm wondering here why do we have these type checks. I thought `FastThreadLocal` only works well with `FastThreadLocalThread` - but you seem to prefer including the `JBossThread` case as well. Why exclude any other thread, but not this kind?

Would be great to add some comment for posterity.

Reply:
also (I'm reading some of its sources) it seems the fallback implementation in `InternalThreadLocalMap` is to use another `static` tlocal when the thread is not implementing `FastThreadLocal` - I'd guess we don't want that to ever be used?

We could add a non-static Tlocal as an additional field in here to be used as fallback?

I can propose that as a follow-up if you prefer - I'm just trying to understand it better.

Reply:
The main reason for limiting to JBossThread and VertxThread is that we control the lifecycle of these threads, so when Quarkus is shutdown those threads will go away. Most of the time we use VertxThread now (even for blocking stuff), so the FastThreadLocal route should be used pretty much every time, with the associated performance increase.